src/main/java/com/example/projectfirst/connector/ConnectorController.java

```diff
@@ -29,7 +28,7 @@ public ConnectorCollection getConnector(@PathVariable(value="id") String id){
     }
 
     @PostMapping("/connectors")
-    public ConnectorCollection postConnector(@RequestBody String yaml) throws IOException {
+    public ConnectorCollection postConnector(@RequestBody String yaml) throws APIPWrongYmlFileOfConnectorException {
```

Comment:
Think about scaling here, having exception this specific it would soon become too wide to handle.  We can be less specific but send a message instead, APIPYamlParsingException("Error while parsing connector from yaml input");